tests/unit/helpers/dates/companiDates.test.js

```diff
@@ -41,11 +50,62 @@ describe('CompaniDate', () => {
   });
 });
 
+describe('GETTER', () => {
+  describe('getUnits', () => {
+    const companiDate = CompaniDatesHelper.CompaniDate('2021-11-24T07:12:08.000Z');
+
+    it('should return units', () => {
+      const result = companiDate.getUnits(['day', 'second']);
+
+      expect(result).toEqual({ day: 24, second: 8 });
+    });
+
+    it('should return empty if unit is plural or invalid', () => {
+      const result = companiDate.getUnits(['days', 'second', 'mois']);
+
+      expect(result).toEqual({ second: 8 });
+    });
+  });
+});
+
 describe('DISPLAY', () => {
   describe('format', () => {
-    let _formatMiscToCompaniDate;
     const companiDate = CompaniDatesHelper.CompaniDate('2021-11-24T07:12:08.000Z');
 
+    it('should return formated date in a string', () => {
+      const result = companiDate.format('\'Le\' cccc dd LLLL y \'à\' HH\'h\'mm \'et\' s \'secondes\'');
+
+      expect(result).toBe('Le mercredi 24 novembre 2021 à 08h12 et 8 secondes');
+    });
+  });
+
+  describe('toDate', () => {
+    const companiDate = CompaniDatesHelper.CompaniDate('2021-11-24T07:00:00.000+01:00');
+
+    it('should return a JSDate equivalent to companiDate (in utc)', () => {
+      const result = companiDate.toDate();
+
+      expect(result).toEqual(new Date('2021-11-24T06:00:00.000Z'));
+    });
+  });
+
+  describe('toISO', () => {
+    const companiDate = CompaniDatesHelper.CompaniDate('2021-12-24T12:00:00.000+03:00');
+
+    it('should return a string ISO 8601 equivalent to companiDate (in utc)', () => {
+      const result = companiDate.toISO();
+
+      expect(result).toEqual('2021-12-24T09:00:00.000Z');
+    });
+  });
+});
+
+describe('QUERY', () => {
+  describe('isBefore', () => {
+    let _formatMiscToCompaniDate;
+    let otherDate = '2021-11-01T10:00:00.000Z';
+    const companiDate = CompaniDatesHelper.CompaniDate('2021-11-01T07:00:00.000Z');
+
```

Comment:
Que penses-tu de definir companiDate et otherDate dans chaque test ? Je trouverais ca plus lisible car ici de temps en temps otherDate est defini dans le it, parfois non. Et ca permettrait de voir au premier coup d'oeil si effectivement les dates sont bien avant/apres comme souhaité dans le test.

Reply:
je suis d'acc

Reply:
Je viens de coder la rectification, j'ai descendu les "let" dans chaque tests (dan chaque "it"), j'ai laissé les "const" au debut des describe.